### PR TITLE
Remove Rack::Sendfile from the default middleware stack

### PR DESCRIPTION
**Pull request by SValkanov**
Discussions: #41148, #42635 

*  `Rack::Sendfile` is replaced by a dummy `FakeSendfile` middleware, which cannot be used in middleware operations
*   Using `Rack::Sendfile` in middleware operations (`insert_after`, `insert_before`, etc.) throws a deprecation warning and uses `FakeSendfile` instead
*   If an application adds `Rack::Sendfile` explicitly (`use`, etc.), then the deprecation warning is not raised and `FakeSendfile` is ignored
*   Removed configuration option for specifying the header used for sending files:
    ```
    config.action_dispatch.x_sendfile_header = 'X-Sendfile'
    ```
*   Docs are updated

### CODE REVIEW

File: actionpack/lib/action_dispatch/middleware/stack.rb [hunk 1]
```diff
@@ -5,6 +5,16 @@
 
 module ActionDispatch
   class MiddlewareStack
+    class FakeSendfile
```
Comment (SkipKayhil) [inline actionpack/lib/action_dispatch/middleware/stack.rb:8]:
```suggestion
    class FakeSendfile # :nodoc:
```

This class should probably not be documented (see e9e8ea2b2071c8228625a72d51e0d3a589611eaa)

**Comment by SValkanov** [comment]
Bumping this for any feedback, @rafaelfranca

**Comment by zarqman** [comment]
Instead of adding `FakeSendfile`, is it possible to just add a config item like `config.action_dispatch.enable_rack_sendfile` (and use it as a conditional at `default_middleware_stack.rb:23`)? In Rails <= 7.0, it would default to `true`, but switch to `false` in >= 7.1. I envision the option being included in `new_framework_defaults_7_1.rb`, which is sometimes how this kind of behavior change is implemented.

If a full-on deprecation is still wanted, then when `true` also add a deprecation message about changing to `use Rack::Sendfile`.

I've also offered thoughts in #41148 on a way to keep Sendfile, although it may still end up being desirable to disable it by default.

Thanks for your work to help make Rails better!

**Comment by SValkanov** [comment]
Friendly ping for any feedback @jonathanhefner

**Pull request closed without merging**